packages/cli/test/utils/simulation/assertions/withdrawalsAssertion.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {capella} from "@lodestar/types";
+import {ApiError} from "@lodestar/api";
+import {MAX_WITHDRAWALS_PER_PAYLOAD} from "@lodestar/params";
+import {AssertionMatch, AssertionResult, SimulationAssertion} from "../interfaces.js";
+
+type WithdrawalsData = {
+  withdrawalCount: number;
+  withdrawalAmount: bigint;
+  validators: Record<number, {balanceInLastSlot: bigint; currentBalance: bigint; withdrawalAmount: bigint}>;
+};
+
+export function createWithdrawalAssertions<T extends string>(
+  nodeId: T
+): SimulationAssertion<`withdrawals_${T}`, WithdrawalsData> {
+  return {
+    id: `withdrawals_${nodeId}`,
+    match({forkConfig, epoch, node}) {
+      if (nodeId === node.id && epoch === forkConfig.CAPELLA_FORK_EPOCH)
+        return AssertionMatch.Capture | AssertionMatch.Assert;
+
+      return AssertionMatch.None;
+    },
+    async capture({block, node, slot}) {
+      const withdrawals = (block as capella.SignedBeaconBlock).message.body.executionPayload.withdrawals;
+      let withdrawalCount = 0;
+      let withdrawalAmount = BigInt(0);
+      const validators: WithdrawalsData["validators"] = {};
+
+      for (const withdrawal of withdrawals) {
```

Comment:
what if it's empty `withdrawals`? is it scanned through `assert()` function below?

Reply:
Yes the assertion works on this logic. 

```js
      if (nodeId === node.id && epoch === forkConfig.CAPELLA_FORK_EPOCH)
        return AssertionMatch.Capture | AssertionMatch.Assert;
```

So even if the above line withdrawals are empty, the assertion will still trigger and results in empty errors array.